src/tracked_struct.rs

```diff
@@ -563,6 +566,25 @@ where
 
         unsafe { self.to_self_ref(&data.fields) }
     }
+
+    /// Access to this untracked field.
+    ///
+    /// Note that this function returns the entire tuple of value fields.
+    /// The caller is responible for selecting the appropriate element.
+    pub fn untracked_field<'db>(
+        &'db self,
+        db: &'db dyn crate::Database,
+        s: C::Struct<'db>,
+        _field_index: usize,
+    ) -> &'db C::Fields<'db> {
+        let (zalsa, _) = db.zalsas();
+        let id = C::deref_struct(s);
+        let data = Self::data(zalsa.table(), id);
+
+        data.read_lock(zalsa.current_revision());
```

Comment:
Is the `read_lock` necessary here? It seems not because `leak_fields` does not call it.

Reply:
Not a 100% but the comment on `Value::updated_at` and the inline comment inside `update` does suggest to me that the lock is required before accessing any field (handing out references to field) and it not being present in `leak_field` seems like a bug.